### ISSUE TITLE
Inline `# rubocop:disable Metrics/ParameterLists` does not suppress offense for multi-line method definitions

### ISSUE THREAD

**Issue by Benjamin-deVries** [rubocop/rubocop#14379]
## Inline `# rubocop:disable Metrics/ParameterLists` does not suppress offense for multi-line method definitions

When using `# rubocop:disable Metrics/ParameterLists` as an **inline comment** at the end of a multi-line method definition, RuboCop does not suppress the offense as expected. Additionally, RuboCop may emit a warning that the disable comment is unnecessary, which is misleading.

This appears to be due to RuboCop only associating inline disable comments with the line they are on, rather than the entire multi-line method definition.

---

#### Working example (single-line method):

```ruby
def self.foo(a:, b:, c:, d:, e:, f:) # rubocop:disable Metrics/ParameterLists
  # ...
end
```

---

#### Not working example (multi-line method):

```ruby
def self.foo(a:, b:, c:,
             d:, e:, f:) # rubocop:disable Metrics/ParameterLists
  # ...
end
```

---

### Expected behavior

The `rubocop:disable` inline comment should suppress the `Metrics/ParameterLists` offense for the method, regardless of whether it spans multiple lines.

---

### Actual behavior

RuboCop still raises the offense and may also show:

```
Unnecessary disabling of Metrics/ParameterLists.
```

---

### Workaround

Using a block-level disable works as expected:

```ruby
# rubocop:disable Metrics/ParameterLists
def self.foo(a:, b:, c:,
                    d:, e:, f:)
  # ...
end
# rubocop:enable Metrics/ParameterLists
```

**Comment by bbatsov** [comment]
Reopened — still reproduces on master (1.88): an inline `# rubocop:disable Metrics/ParameterLists` on the closing-paren line of a multi-line `def` isn't honored (the offense is still reported), whereas the same disable on a single-line method works.
